src/Runner.Worker/Handlers/ScriptHandlerHelpers.cs

```diff
@@ -79,5 +81,22 @@ internal static (string shellCommand, string shellArgs) ParseShellOptionString(s
                 throw new ArgumentException($"Failed to parse COMMAND [..ARGS] from {shellOption}");
             }
         }
+
+        internal static string WhichShell(string path, Common.Tracing trace, string prependPath) // TODO: Figure out how it relates to 
+        {
+            var format = "{0} {1}";
+            switch (Path.GetExtension(path))
+            {
+                case ".sh":
+                    // use 'sh' args but prefer bash
+                    var pathToShell = WhichUtil.Which("bash", false, trace, prependPath) ?? WhichUtil.Which("sh", true, trace, prependPath);
+                    return string.Format(format, pathToShell, _defaultArguments["sh"]);
+                case ".ps1":
+                    var pathToPowershell = WhichUtil.Which("pwsh", false, trace, prependPath) ?? WhichUtil.Which("powershell", true, trace, prependPath);
+                    return string.Format(format, pathToPowershell, _defaultArguments["powershell"]);
+                default:
+                    throw new ArgumentException($"{path} is not a valid path to a script. Make sure it ends in '.sh' or '.ps1'.");
```

Comment:
I must admit that this tripped me up, even after our discussions. I expected to be able to `ACTIONS_RUNNER_HOOK_JOB_STARTED` to an sh script that _didn't_ end in `.sh` and have it Just Work.

We don't need to block this PR on this, but I think that we should fallback to `sh -c {path}` on POSIX instead of throwing here.

Reply:
We can always become more permissive in the future here